kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/kms/aws/AwsKmsLocal.java

```diff
@@ -0,0 +1,187 @@
+                            .exec("sh", "-c", "curl " + url + "/_" + LOCALSTACK_SERVICE_NAME + "/info")) {
+                int exitCode = exec.exitCode().join();
+                if (exitCode != 0) {
+                    throw new UnsupportedOperationException(error.toString());
+                }
+                // version returned with format: "3.5.1.dev:6e7ddd05e"
+                installedLocalStackVersion = new ObjectMapper().readTree(output.toString()).findValue("version").textValue().split(":")[0];
+            }
+            catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+
+        return installedLocalStackVersion;
+    }
+
+    @Override
+    public void deploy() {
+        LOGGER.info("Deploy AWS in {} namespace", deploymentNamespace);
+        if (isDeployed()) {
+            LOGGER.warn("Skipping AWS deployment. It is already deployed!");
+            return;
+        }
+
+        NamespaceUtils.createNamespaceWithWait(deploymentNamespace);
+        ResourceManager.helmClient().addRepository(LOCALSTACK_HELM_REPOSITORY_NAME, LOCALSTACK_HELM_REPOSITORY_URL);
+        ResourceManager.helmClient().namespace(deploymentNamespace).install(LOCALSTACK_HELM_CHART_NAME, LOCALSTACK_SERVICE_NAME,
+                Optional.of(Environment.AWS_LOCALSTACK_CHART_VERSION),
+                Optional.of(Path.of(TestUtils.getResourcesURI("helm_localstack_overrides.yaml"))),
+                Optional.empty());
+
+        this.podName = KafkaUtils.getPodNameByLabel(deploymentNamespace, "app.kubernetes.io/name", LOCALSTACK_SERVICE_NAME, Duration.ofSeconds(30));
+
+        if (!isCorrectVersionInstalled()) {
+            throw new KubeClusterException("AWS version installed " + getLocalStackVersionInstalled() + " does not match with the expected: '"
+                    + AwsKmsTestKmsFacade.LOCALSTACK_IMAGE.getVersionPart() + "'");
+        }
+    }
+
+    private boolean isCorrectVersionInstalled() {
+        String installedVersion = getLocalStackVersionInstalled();
+        String expectedVersion = AwsKmsTestKmsFacade.LOCALSTACK_IMAGE.getVersionPart();
+
+        return TestUtils.compareInstalledVersions(installedVersion, expectedVersion) == 0;
+    }
+
+    @Override
+    public void delete() {
+        LOGGER.info("Deleting Aws in {} namespace", deploymentNamespace);
+        NamespaceUtils.deleteNamespaceWithWait(deploymentNamespace);
+    }
+
+    @Override
+    public URI getAwsUrl() {
+        var nodeIP = kubeClient(deploymentNamespace).getClient().nodes().list().getItems().get(0).getStatus().getAddresses().get(0).getAddress();
+        var spec = kubeClient().getService(deploymentNamespace, LOCALSTACK_SERVICE_NAME).getSpec();
+        int port = spec.getPorts().stream().map(ServicePort::getNodePort).findFirst()
+                .orElseThrow(() -> new KubeClusterException("Unable to get the service port of Aws"));
+        String url = nodeIP + ":" + port;
```

Comment:
AFAIK, we need to get the IP and Port to create the URL for the Kroxylicious configMap. To access `aws` in localstack we need to point to the proper IP and NodePort.
In addition it is the recommendation `Localstack` give us when installing it with the Helm Chart:
```
$ helm install localstack localstack/localstack --wait
NAME: localstack
LAST DEPLOYED: Wed Jul  3 07:19:57 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
NOTES:
1. Get the application URL by running these commands:
  export NODE_PORT=$(kubectl get --namespace "default" -o jsonpath="{.spec.ports[0].nodePort}" services localstack)
  export NODE_IP=$(kubectl get nodes --namespace "default" -o jsonpath="{.items[0].status.addresses[0].address}")
  echo http://$NODE_IP:$NODE_PORT
```

Reply:
I think this impl queries for the first node in the cluster not the first node running the localstack instance. 

I could be miss reading things on my phone

Reply:
This [page](https://docs.localstack.cloud/user-guide/integrations/kubernetes/) makes it look like localstack uses a `NodePort` service.  If that's the case getting the IP the first node is a reasonable approach.    We just need to be certain it will be routable in all environments where the tests will be run.